### PR TITLE
Add CreateOrder post endpoint, integration tests, and few fixes

### PR DESCRIPTION
**Pull request by s21166-pj**
## Add CreateOrder post endpoint, integration tests, and few fixes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

- [ ] Unit test(s)
- [x] Integration test(s)
- [x] Tested manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings
- [X] I have added frontend code and checked it for mobile & desktop

### CODE REVIEW

File: src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        public async Task<CreateOrderCreateOrderResponse> CreateOrder(CreateOrderCreateOrderPayload payload)
+        {
+            var auth0UserInfo = await _currentUserService.GetUserInfo();
+            var currentUser = await _getCurrentUserService.Execute(auth0UserInfo);
+
+            await _roleService.ThrowIfNoRole(RoleName.Needy, currentUser.Id);
+            await _createOrderCreateOrderPayloadValidator.ValidateAndThrowAsync(payload);
+
+            var listOfProducts = await _appDbContext.Products.Where(c => c.Status == DbEntityStatus.Active).FromCache().ToDynamicListAsync();
```
Comment (h4570) [inline src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs:58]: Yyy ale po co to pobierasz?

File: src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            var listOfOrderProduct = payload.Products.Select(c => new OrderProduct
+            {
+                ProductId = _hashids.DecodeSingleLong(c.Id),
+                Quantity = c.Quantity,
+                UpdateUserId = currentUser.Id,
+                UpdatedAt = DateTime.UtcNow,
+                Status = DbEntityStatus.Active
+            }).ToList();
+
+            var nextOrderId = _counterService.GetAndUpdateNextCounter(TableName.Orders);
```
Comment (h4570) [inline src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs:74]: Await ................

Przejrzyj kod zanim mi go wyślesz..

**Review by h4570 — CHANGES_REQUESTED** [review]

File: src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderResponse.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Orders.Actions.CreateOrderActions.CreateOrderCreateOrder
+{
+    public class CreateOrderCreateOrderResponse
+    {
+        public string OrdereName { get; set; } = default!;
```
[the remark this answers is not in the document]
Comment (s21166-pj) [reply src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderResponse.cs:5]: hehe

File: src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs (hunk 1 above)
Comment (s21166-pj) [reply src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs:58]: Yep, that's on me

File: src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderPayloadValidator.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using FluentValidation;
+
+namespace Orders.Actions.CreateOrderActions.CreateOrderCreateOrder
+{
+    public class CreateOrderCreateOrderPayloadValidator : AbstractValidator<CreateOrderCreateOrderPayload>
+    {
+        public CreateOrderCreateOrderPayloadValidator()
+        {
+            RuleFor(c => c.AddressId).NotNull().Length(1, 50);
```
[the remark this answers is not in the document]
Comment (s21166-pj) [reply src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderPayloadValidator.cs:9]: Dodany fix

File: src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs (hunk 2 above)
Comment (s21166-pj) [reply src/backend/Order/Actions/CreateOrderActions/CreateOrderCreateOrder/CreateOrderCreateOrderService.cs]: mb, pociągnięte z PerformDonation i tam też update'nałem awaita

**Review by h4570 — APPROVED** [review]

**Pull request merged**